diff --git a/include/configs/mt8370.h b/include/configs/mt8370.h
index 06ccd71bb6..ba96aa81ec 100644
--- a/include/configs/mt8370.h
+++ b/include/configs/mt8370.h
@@ -92,6 +92,8 @@
 	"splashfile=logo.bmp\0" \
 	"splashdevpart=0#bootassets\0" \
 	"splashpos=m,m\0" \
+	"bootargs=root=PARTLABEL=rootfs rootwait video=HDMI-A-1:800x480@60\0" \
+	"hdmi_1024x600=setenv bootargs root=PARTLABEL=rootfs rootwait video=HDMI-A-1:1024x600@60\0" \
 	BOOTENV
 #else
 #define CONFIG_EXTRA_ENV_SETTINGS \
@@ -105,6 +107,8 @@
 	"splashimage=" __stringify(CONFIG_SYS_LOAD_ADDR) "\0" \
 	"splashsource=sf\0" \
 	"splashpos=m,m\0" \
+	"bootargs=root=PARTLABEL=rootfs rootwait video=HDMI-A-1:800x480@60\0" \
+	"hdmi_1024x600=setenv bootargs root=PARTLABEL=rootfs rootwait video=HDMI-A-1:1024x600@60\0" \
 	BOOTENV
 #endif
 #endif
